### PR TITLE
Restructure packaging for components-core

### PR DESCRIPTION
**Pull request by OscarSpruit**
## Description
Good luck reviewing this one 😬 

image

COAND-497

### CODE REVIEW

File: components-core/src/main/java/com/adyen/checkout/components/core/internal/provider/ActionComponentProvider.kt [hunk 1]
```diff
@@ -15,12 +15,15 @@ import androidx.lifecycle.LifecycleOwner
...
-import com.adyen.checkout.components.model.payments.response.Action
+import com.adyen.checkout.components.core.action.Action
+import com.adyen.checkout.components.core.internal.ActionComponent
+import com.adyen.checkout.components.core.internal.ActionComponentCallback
+import com.adyen.checkout.components.core.internal.Component
+import com.adyen.checkout.components.core.internal.Configuration
+import com.adyen.checkout.components.core.internal.ui.ActionDelegate
 
 // TODO SESSIONS: DOCS
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
```
Comment (etman55) [inline components-core/src/main/java/com/adyen/checkout/components/core/internal/provider/ActionComponentProvider.kt:26]: I guess we don't need `@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)` on `getDelegate()` method anymore after adding this to the whole interface

File: components-core/src/main/java/com/adyen/checkout/components/core/internal/ui/ActionDelegate.kt [hunk 2]
```diff
@@ -6,16 +6,18 @@
  * Created by josephj on 24/8/2022.
  */
 
-package com.adyen.checkout.components.base
+package com.adyen.checkout.components.core.internal.ui
```
Comment (etman55) [inline components-core/src/main/java/com/adyen/checkout/components/core/internal/ui/ActionDelegate.kt:9]: hmmm I am thinking this should be inside action module WDYT?

File: components-core/src/main/java/com/adyen/checkout/components/core/internal/provider/ActionComponentProvider.kt [hunk 3]
```diff
@@ -5,7 +5,7 @@
  *
  * Created by caiof on 13/5/2019.
  */
-package com.adyen.checkout.components
+package com.adyen.checkout.components.core.internal.provider
```
Comment (etman55) [inline components-core/src/main/java/com/adyen/checkout/components/core/internal/provider/ActionComponentProvider.kt:8]: same here, should this be inside action module?

File: components-core/src/main/java/com/adyen/checkout/components/core/internal/ui/ViewableDelegate.kt [hunk 4]
```diff
@@ -6,10 +6,13 @@
  * Created by josephj on 24/8/2022.
  */
 
-package com.adyen.checkout.components.base
+package com.adyen.checkout.components.core.internal.ui
```
Comment (etman55) [inline components-core/src/main/java/com/adyen/checkout/components/core/internal/ui/ViewableDelegate.kt:9]: I think this should be inside ui-core module WDYT?

File: components-core/src/main/java/com/adyen/checkout/components/core/internal/provider/ActionComponentProvider.kt (hunk 1 above)
Comment (OscarSpruit) [reply components-core/src/main/java/com/adyen/checkout/components/core/internal/provider/ActionComponentProvider.kt:26]: I'd like to keep it there to make extra clear that it shouldn't be used

File: components-core/src/main/java/com/adyen/checkout/components/core/internal/ui/ViewableDelegate.kt (hunk 4 above)
Comment (OscarSpruit) [reply components-core/src/main/java/com/adyen/checkout/components/core/internal/ui/ViewableDelegate.kt:9]: I think we should leave it here. `ui-core` is more "real" UI related, so custom views and styling. There are some not so UI related classes in there, but let's try and keep that to a minimum

**Review by etman55 — APPROVED** [review]

**Pull request merged**